helix-term/src/ui/mod.rs

```diff
@@ -265,6 +266,69 @@ pub fn file_picker(root: PathBuf, config: &helix_view::editor::Config) -> FilePi
     picker
 }
 
+pub fn file_browser(root: PathBuf) -> FilePicker {
+    let directory_content = directory_content(&root);
+
+    let columns = [PickerColumn::new(
+        "path",
+        |item: &PathBuf, root: &PathBuf| {
+            item.strip_prefix(root)
+                .unwrap_or(item)
+                .to_string_lossy()
+                .into()
+        },
+    )];
+    let picker = Picker::new(columns, 0, [], root, move |cx, path: &PathBuf, action| {
+        if path.is_dir() {
+            let owned_path = path.clone();
+            let callback = Box::pin(async move {
+                let call: Callback =
+                    Callback::EditorCompositor(Box::new(move |_editor, compositor| {
+                        let picker = file_browser(owned_path);
+                        compositor.push(Box::new(overlay::overlaid(picker)));
+                    }));
+                Ok(call)
+            });
+            cx.jobs.callback(callback);
+        } else if let Err(e) = cx.editor.open(path, action) {
+            let err = if let Some(err) = e.source() {
+                format!("{}", err)
+            } else {
+                format!("unable to open \"{}\"", path.display())
+            };
+            cx.editor.set_error(err);
+        }
+    })
+    .with_preview(|_editor, path| Some((path.as_path().into(), None)));
+    let injector = picker.injector();
+
+    if let Ok(files) = directory_content {
+        for file in files {
+            if injector.push(file).is_err() {
+                break;
+            }
+        }
+    }
+    picker
+}
+
+fn directory_content(path: &Path) -> Result<Vec<PathBuf>, std::io::Error> {
```

Comment:
To be able to ascend up a directory we could add `..` to this list whenever the path isn't the root

Reply:
Nice idea and was quite easy to implement too, thanks! Should have added it from the start, really =)